mayastor/src/rebuild.rs

```diff
@@ -392,29 +449,32 @@ impl RebuildOperations for RebuildJob {
         let destination = self.destination.clone();
         let complete_receiver = self.complete_chan.clone().1;
 
-        Reactors::current().send_future(async move {
+        Reactors::get_by_core(0).unwrap().send_future(async move {
             let job = match RebuildJob::lookup(&destination) {
                 Ok(job) => job,
                 Err(_) => {
                     return error!(
-                        "Failed to find the rebuild job {}",
+                        "Failed to find and start the rebuild job {}",
                         destination
                     );
                 }
             };
 
-            job.run().await;
+            // todo: WA until cas-194 is addressed...
+            if job.state == RebuildState::Pending {
+                job.run().await;
+            }
         });
         complete_receiver
     }
 
     fn stop(&mut self) -> Result<(), RebuildError> {
         match self.state {
-            RebuildState::Paused => {
+            RebuildState::Pending | RebuildState::Paused => {
                 self.change_state(RebuildState::Stopped);
                 // The rebuild is paused so call complete here
                 // because the run function is inactive
```

Comment:
The comment needs updating to accommodate the Pending state.

Reply:
Done